tests/MongoDB.Driver.Examples/CausalConsistencyExamples.cs

```diff
@@ -0,0 +1,77 @@
+﻿/* Copyright 2018-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using Xunit;
+
+namespace MongoDB.Driver.Examples
+{
+    public class CausalConsistencyExamples
+    {
+        [Fact]
+        public void Causal_Consistency_Example_2()
+        {
+            RequireServer.Check().SupportsCausalConsistency();
+
+            PrepareClient();
+
+            // Start Tunable consistency controls example
+            var connectionString = "mongodb://localhost/?readPreference=secondaryPreferred";
+
+            var client = new MongoClient(connectionString);
+            var database = client.GetDatabase("myDatabase");
+            var collection = database.GetCollection<BsonDocument>("myCollection");
+
+            // Start client session, which is causally consistent by default
+            using (var session = client.StartSession())
+            {
+                // Run causally related operations within the session
+                collection.InsertOne(session, new BsonDocument("name", "Tom"));
+                collection.UpdateOne(session,
+                    Builders<BsonDocument>.Filter.Eq("name", "Tom"),
+                    Builders<BsonDocument>.Update.Set("name", "John"));
+
+                foreach (var document in collection.Find(session, FilterDefinition<BsonDocument>.Empty).ToEnumerable())
+                {
+                    // process document
+                }
+            }
+            // End Tunable consistency controls example
+
+            var result = collection.Find("{}").FirstOrDefault();
+            RemoveIds(new[] { result });
+            result.Should().Be("{ name: 'John' }");
+        }
+
+        private IMongoClient PrepareClient()
```

Comment:
This was not what I suggested, but oh well...

Reply:
This violates the single responsibility principle. This method now does two unrelated things:

1. Initialize a client instance
2. Drop a collection

Reply:
I suggest splitting this into two methods:

1. CreateClient
2. DropCollection

Reply:
Sorry, I didn't understand you correctly previously, I will update this part